### DISCUSSION TITLE
External DLL support in Mojo

### DISCUSSION THREAD

**Discussion by sune-gaulsh** [modular/modular#508] [Ideas]
It would be really awesome if Mojo supported calling functions from external dlls via external function pointers and passing pointers to preallocated memory.  I am thinking C/C++/Fortran etc.

**Comment by lattner** [comment]
This should already work, but it is super low level.  This is how we talk to the CPython interpreter for example.

**Comment by abduld** [comment]
@sune-gaulsh . Take a look at the external_call function (https://docs.modular.com/mojo/MojoStdlib/Intrinsics.html#external_call)
